dist/mushroom-strategy.js

```diff
@@ -45,55 +48,52 @@ function getDeviceEntitiesFromRegistry(entities, devices, area, startsWith) {
  * @param {string} startsWith Starting string of the entity-id.
  *
  * @return {Set<stateObject>} Set of state entities.
+ * @todo: Apply a filter to stateEntities instead of iterating it manually.
  */
-const getFilteredStates = (info, entities, devices, area, startsWith) => {
-  const entityLookup = Object.fromEntries(entities.map((ent) => [ent.entity_id, ent]));
-  const deviceLookup = Object.fromEntries(devices.map((dev) => [dev.id, dev]));
-
-  let states = Object.values(info.hass.states).filter(
-      (stateObj) => stateObj.entity_id.startsWith(startsWith),
+function getStateEntities(info, entities, devices, area, startsWith) {
+  const states = new Set;
+
+  // Create a map for the hassEntities and devices {id: object} to improve lookup speed.
+  /** @type {Object<string, hassEntity>} */
+  const entityMap = Object.fromEntries(entities.map(entity => [entity.entity_id, entity]));
+  /** @type {Object<string, deviceEntity>} */
+  const deviceMap = Object.fromEntries(devices.map(device => [device.id, device]));
+
+  // Get states whose entity-id starts with the given string.
+  const stateEntities = Object.values(info.hass.states).filter(
+      state => state.entity_id.startsWith(startsWith),
   );
 
-  const areaEntities = new Set;
-
-  for (const stateObj of states) {
-    const entry = entityLookup[stateObj.entity_id];
-
-    if (!entry) {
-      continue;
-    }
-
-    if (entry.area_id) {
-      if (entry.area_id !== area.area_id) {
-        continue;
-      }
-    } else if (entry.device_id) {
-      const device = deviceLookup[entry.device_id];
+  for (const state of stateEntities) {
+    const hassEntity = entityMap[state.entity_id];
+    const device     = deviceMap[hassEntity.device_id];
```

Comment:
When running opening a view it gives error 
`TypeError: Cannot read properties of undefined (reading 'device_id')`
not all `hassEntity` has `device_id`

Reply:
> not all hassEntity has device_id

Actually, that's what I expected, but for some reason, this error never raised at my instance.
Hence the removal of the check for the device value-type.

Could you give an example of an entity without a device, which I probably have as well?
I just want to double-check.

Reply:
I have some template entities specifically fan templates, template sensors, and cameras that don't have device IDs.  You can try adding a quick template light/fan/sensor/etc

Reply:
Which client do you use?
Over here, the following statement evaluates to `undefined` without throwing any errors.
```js
>> device = deviceMap[hassEntity.ferry_id]
<- undefined
```

Reply:
I've been able to recreate the situation.
It's about entities which don't have a `unique_id` attribute (Sensor, binary sensor, button, number and select template entities which follow the modern configuration format).
See https://www.home-assistant.io/integrations/template/#state-based-template-binary-sensors-buttons-numbers-selects-and-sensors.

These entities aren't added to the entity registry.
For them `hassEntity` is `undefined` and trying to read a property of it, raises the error.

I'm pretty sure you can't assign them to an area either, so they can safely be skipped for a view of area-related entities.

It does make me think, though...
Mushroom-strategy by default, only supports views for the defined areas and views for domains lights, switches, covers, cameras and thermostats.
This means, by default, above entities are nowhere included in the dashboard.
Should we have to create some kind of "More..." view for them?

Reply:
I see, 
As far as I know, my instance has those template entities but I provided a unique ID.
When giving the unique ID then I can add them to my areas and use them in the dashboard.
I did think of adding more entities view or smth similar but I figured if someone really wants to add an entity with an unassigned area then they can use extra cards and other similar options. Feel free to give a counterargument

Reply:
> I see, As far as I know, my instance has those template entities but I provided a unique ID. When giving the unique ID then I can add them to my areas and use them in the dashboard.

My bad, I was under the impression that you couldn't add a unique_id when using the modern format.
After re-reading the [docs](https://www.home-assistant.io/integrations/template/#all-sensor-binary-sensor-button-number-select-entities), it seems you can after all.

> I did think of adding more entities view or smth similar but I figured if someone really wants to add an entity with an unassigned area then they can use extra cards and other similar options. Feel free to give a counterargument

Having to add "missing" entities manually, kind of defeats the purpose of an auto-generated dashboard. I think somehow including all and implementing #10 is a finer way to go.
Now we have implicit exclusion, then we'll have explicit exclusion.
`quick_access_cards` and `extra_cards` can/should still be used for including custom cards.

Reply:
> Do you know which devices don't have unique ids?

I think we don't have to worry about the `unique_id`. Like I already admitted, I've misread the docs. 😌
Probably all entities could/should have a `unique_id` attribute.

> It seems that the more view wouldn't be used as much...

Probably not as much, but in my case I have dozens of entities which do not belong to any of the current views or areas (sub views).
Even though not used as much as other entities, many are still important to me.

The current views are covering domains:
- light
- fan
- cover
- switch
- climate
- cameras

The sub views are covering all domains of a specific area.

So, having an entity of the domain sensor which doesn't have an area specified (e.g. hacs pending updates), is nowhere to be found in the generated dashboard.

I' still prefer explicit exclusion and appreciate an additional view to cover the entities which are now missing, like...
![afbeelding](https://user-images.githubusercontent.com/7613487/232321520-8ee2c445-96f1-4ba7-a1a3-8ab5605e3ed1.png)

Reply:
Reading the third paragraph [here](https://www.home-assistant.io/faq/unique_id/), it seems that almost all YAML entities don't have `unique_id`. Only some template entities have a `unique _id` field

I can see a bit more of the use case. Since there is a need for explicit exclusion adding this view makes sense. Do you prefer it to enabled or disabled by default?

Reply:
According to the [Template documentation](https://www.home-assistant.io/integrations/template/#unique_id), you can define a `unique_id` for all sensor, binary sensor, button, number, select entities. 🙄

Welcome to the exploration of other peoples code and documentation. 😁
I do hope you enjoy the challenge.

Reply:
The actual issue is resolved.
Maybe we can continue this discussion somewhere else, so the conversation can be resolved also.

One options is to [enable discussions](https://docs.github.com/en/discussions/quickstart) for the repository.
Otherwise we should exchange contact details for some more privacy.